### ISSUE TITLE
Breakpoint in .pm file attempts to open module file located in cwd

### ISSUE THREAD

**Issue by jcloggins** [Komodo/KomodoEdit#346]
When a breakpoint in a .pm module is encountered while debugging, Komodo attempts to open the module file in the cwd instead of its true location.

Set up:
Create a .pm module and a .pl file that uses it
Set a breakpoint in the .pm module
In the .pl file, chdir to set the cwd
Call the function the .pm file that contains the breakpoint

Note that you get a popup saying 
The URI ‘file://///Comupter Network Name/… cwd …/ModuleName.pm could not be opened

Also note that a breakpoint has been created in the cwd/ModuleName.pm file that can never be deleted.

I am using:
Komodo IDE, version 9.1.0, build 87337, platform win32-x86.
Built on Fri May 29 18:45:28 2015.

with perl 5.20.2 on Windows 7

I  don’t see anything relevant to this in the log file

[2015-06-16 11:48:30,917] [INFO] Startup: Welcome to Komodo IDE 9.1.0 build 87337 (platform win32-x86, running on Windows 7 version 6.1.7601)
[2015-06-16 11:48:30,917] [INFO] Startup: C:\Program Files\ActiveState Komodo IDE 9\lib\mozilla\komodo.exe built on Fri May 29 18:45:28 2015
[2015-06-16 11:48:31,861] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2015-06-16 11:48:31,861] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'component' directive. (1) in file:///C:/Program%20Files/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:1
[2015-06-16 11:48:31,861] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'contract' directive. (1) in file:///C:/Program%20Files/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:2
[2015-06-16 11:48:31,862] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'interfaces' directive. (1) in file:///C:/Program%20Files/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:3
[2015-06-16 11:48:31,926] [WARNING] root: ko.logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:15:1

[2015-06-16 11:48:32,331] [WARNING] console-logger: Unknown property 'border-radius-top-left'.  Declaration dropped. (1) in less://commando/skin/commando.less:19
[2015-06-16 11:48:32,332] [WARNING] console-logger: Unknown property 'border-radius-top-right'.  Declaration dropped. (1) in less://commando/skin/commando.less:20
[2015-06-16 11:48:32,772] [ERROR] console-logger: bad script XDR magic number (0) in :0
Traceback (most recent call last):
  File "", line 0, in 

[2015-06-16 11:48:35,855] [WARNING] root: ko.findresults was already loaded, re-creating it.

[2015-06-16 11:48:35,948] [DEBUG] places_js: waiting for document complete
[2015-06-16 11:48:38,683] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/komodo.xul:0
[2015-06-16 11:48:38,686] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/komodo.xul:0
[2015-06-16 11:48:39,532] [ERROR] console-logger: bad script XDR magic number (0) in :0
Traceback (most recent call last):
  File "", line 0, in 

[2015-06-16 11:48:39,779] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///C:/Program%20Files/ActiveState%20Komodo%20IDE%209/lib/mozilla/components/koamAddonManager.js:138
[2015-06-16 11:54:48,102] [ERROR] console-logger: NS_ERROR_FILE_NOT_FOUND: Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [koIAppInfoEx.localHelpFile] (2) in chrome://komodo/content/browse.js:185
Traceback (most recent call last):
  File "chrome://komodo/content/browse.js", line 185, in 

[2015-06-16 11:58:01,868] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/tail/tail.xul:0
[2015-06-16 11:58:01,869] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/tail/tail.xul:0

**Comment by jcloggins** [comment]
Follow-up on the un-removable breakpoint: I did find a way to delete the breakpoint. You have to select it in the breakpoint window and use the “Delete” key.

The Komodo’s Delete Breakpoint and Clear All Breakpoints functions still don’t get the job done.

**Comment by cgchoffman** [comment]
The stuck breakpoint is issue #256, likely.

Are you doing remote debugging by any chance?

**Comment by jcloggins** [comment]
While I certainly wouldn’t rule out the possibility that the two breakpoint issues are closely related, they have _very_ different symptoms.

In #256, the breakpoint can be deleted from the breakpoint pane via the “Clear all breakpoints” command, but the red dot indicator remains in the editor pane. If you exit Komodo and then reopen it, the breakpoint is reestablished in the break point pane – at which time it can be cleanly deleted.

In this instance, the “Clear all breakpoints” command does not remove the breakpoint from the breakpoint pane. In this case, you also get a popup about not being able to open the file the breakpoint points to (i.e. the file in the cwd).

As to the question of remote debugging: 

No, I was not. Nor did the chdir reference a networked drive. The .pl and .pm file were in the project directory on the C: drive of my PC and the cwd after the chdir was a subdirectory of the project directory.

**Comment by cgchoffman** [comment]
Please check for and remove any Mapped URIs you might have in your preferences. Edit > Preferences > Mapped URIs

**Comment by jcloggins** [comment]
None were defined

![uri](https://cloud.githubusercontent.com/assets/12155798/8214110/3cc2fe7c-14e4-11e5-9063-5704e01d0a7e.jpg)

**Comment by jcloggins** [comment]
Here is some example code

Note: the project folder needs to contain a subfolder named "NewCWD"

BreakPointTest.pl
***********************************************
#!/usr/bin/perl -w
use strict;
use BreakPointModule;

my $target = BreakPointModule->new;

chdir "NewCWD";

$target->BreakpointTarget();
***********************************************

BreakPointModule.pm
***********************************************
package BreakPointModule;
use strict;
use warnings;

require Exporter;
our @ISA = qw(Exporter);
our @EXPORT_OK = qw();

sub new
{
   my $class = shift;
   my $self = {};
   bless $self, $class;
   return $self;
}

sub BreakpointTarget()
{
   print "The breakpont goes here"
}

1;
***********************************************

BreakPointTest.komodoproject
***********************************************
<?xml version="1.0" encoding="UTF-8"?>

<project id="ae457bcd-5c33-4359-8eea-117153e49fb5" kpf_version="5" name="BreakPointTest.komodoproject">
<preference-set idref="ae457bcd-5c33-4359-8eea-117153e49fb5" id="project" preftype="project">
</preference-set>
</project>
***********************************************

**Comment by jcloggins** [comment]
Looks like the bulk of the project file got truncated in transmission, but it was the most basic version possible

**Comment by jcloggins** [comment]
#346 is tagged as "Pending Response" but I'm not sure what response is expected at this point.

**Comment by Naatan** [comment]
You need to set up mapped URI's if you want Komodo to automatically resolve the project file location of your files. Please see the documentation: http://docs.activestate.com/komodo/9.2/prefs.html#mapped_uris

**Comment by jcloggins** [comment]
OK a little confusion here since the previous comment was to remove URI mapping.

However, the fix (attachment) was only partially successful - note that 509-JLOGGINS7 is the network name for the local computer. 

It got rid of the pop-up and the read only file. But now there is a different problem: the breakpoint location is not displayed. The file in which the breakpoint occurs does not open and if the file is opened manually the yellow arrow indicating the breakpoint location is not displayed.

In short, if you have more than one breakpoint, you have no clue as to which one you hit.

One thing that I noticed in the documentation: it is for a Unix style system. Has this functionality been verified on Windows 7?

![uri](https://cloud.githubusercontent.com/assets/12155798/10313958/7c1e96bc-6c0f-11e5-9688-a25eb2ec1f08.jpg)

**Comment by tonyattwood** [comment]
Hi,

I've  recently had absolutely new experience, you've got to see that! Just take  a  look here <http://excellent.jewmocracy.com/e4but>


Later, tony.attwood@ntlworld.com
